CHANGES.md

```diff
@@ -1,5 +1,124 @@
 # Changes
 
+## stable-2.12.0
+
+This release introduces route-based policy to Linkerd, allowing users to define
+and enforce authorization policies based on HTTP routes in a fully zero-trust
+way. These policies are built on Linkerd's strong workload identities, secured
+by mutual TLS, and configured using types from Kubernetes's [Gateway API](https://gateway-api.sigs.k8s.io/).
+
+The 2.12 release also introduces access logging, a long-awaited feature that
+optionally allows Linkerd to produce Apache-style request logs, optional
+support for `iptables-nft`, and a host of other improvements and performance
+enhancements.
+
+**Upgrade notes**: Please see the [upgrade instructions](upgrade-2120). <!-- TODO fix link -->
+
+* Proxy
```

Comment:
A couple of changes that I remember, curious if folks think they should be added in?

* non-http traffic forwarding within the mesh when running in `ingress mode` [`edge-22.5.2`](https://github.com/linkerd/linkerd2/blob/main/CHANGES.md#edge-2252) (feels to me like it has been requested by some people, although some time passed since we shipped it so my memory's a bit hazy)
* log stream endpoint (https://github.com/linkerd/linkerd2/issues/7250)
* `process_uptime_seconds_total` to keep track of the number of seconds since the proxy started (added in `edge-22.5.2`) (imo people who want to monitor performance, or debug, might benefit from being aware this is a thing).

Also, we mention access logging in the overview but we do not have a bullet point for it, should it be added in as well?

cc: @hawkw @olix0r